src/rounds/game_modes/empty.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+from arena.arena import Arena
+from player import Player
+from rounds.game_modes.game_mode import GameMode
+from sprite_lists import SpriteLists
+
+# from player_manager import PlayerManager
+
+if TYPE_CHECKING:
+    from player import Player
+
+
+class EmptyGameModePlayerState:
+    def __init__(self, player: Player, lives: int):
+        self.lives = lives
+        self.player = player
+
+
+class EmptyGameMode(GameMode):
+    def __init__(self, players: list[Player]):
+        super().__init__()
+        self.players = players
+
+        for player in players:
+            player.game_mode_state = EmptyGameModePlayerState(player, lives=1)
+
+    def on_round_init(self, arena: Arena, sprite_lists: SpriteLists):
+
+        super().on_round_init(arena, sprite_lists)
+
+    def on_round_start(self):
+        pass
+
+    def create_hud(self):
+        pass
+
+    def create_player_hud(self, player: Player):
+        pass
+
+    def update(self, delta_time: float):
+        pass
+
+    def on_player_death(self, player: Player):
+        state = player.game_mode_state
+        state.lives = 1
+        player.allowed_to_respawn = state.lives > 0
```

Comment:
I think this should be removed?  It's unclear what this method is doing, and I bet it's some copy-paste from `StockGameMode`.  It sets lives to 1, but lives is already 1 since this game mode never changes the number of lives.  The `state.lives > 0` will always be true.


To be honest, I still think it's a bit weird to be showing a heart in a mode where you can't die?  Seems like it would confuse the player.  Showing one heart for everyone makes it seem like sudden death.  I dunno, lemme know what you think.

Reply:
I can understand that. I'm not married to it, I think when putting it in, I had this idea to show 'a heart' = 'infinity symbol'(i have a picture in my head from smash bros.) and I also wanted to write something that was adaptable, would work with stock and another mode with empty being the next mode we approached

Reply:
The infinity sign makes sense, then it's actually conveying useful and accurate information to the player.

At the very least, we can remove this logic that's not doing anything.  This `def on_player_death` method override isn't actually doing anything, everything still behaves identically if we remove it.